### PR TITLE
VS Code launch configurations.

### PR DESCRIPTION
**Pull request by MrTarantula**
This will create (or overwrite) the `launch.json` file for VS Code, with debug, profile, and release modes for each flavor.

I was unsure of if you want it to be a default processor or how to set up the folder structure. Let me know if you want anything changed.

### CODE REVIEW

**Comment by AngeloAvv** [comment]
Sounds good, since the json file is so small and not so long, I wouldn't have used the CopyFileProcessor, but it's fine.

> I was unsure of if you want it to be a default processor or how to set up the folder structure. Let me know if you want anything changed.

I think it's better to think more about it. For instance I don't use VSCode, but Android Studio instead.

File: lib/processors/ide/vscode_launch_processor.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import 'package:flutter_flavorizr/parser/models/flavors/flavor.dart';
+import 'package:flutter_flavorizr/processors/commons/copy_folder_processor.dart';
+import 'package:flutter_flavorizr/processors/commons/queue_processor.dart';
+import 'package:flutter_flavorizr/processors/commons/replace_string_processor.dart';
+import 'package:flutter_flavorizr/processors/commons/runtime_file_string_processor.dart';
+
+class VSCodeLaunchProcessor extends QueueProcessor {
+  VSCodeLaunchProcessor(
+      String source, String destination, Map<String, Flavor> flavors)
```
Comment (AngeloAvv) [inline lib/processors/ide/vscode_launch_processor.dart:9]: You can pass directly the flavors.keys from outside

File: lib/processors/processor.dart [hunk 2]
```diff
@@ -76,6 +77,9 @@ class Processor extends AbstractProcessor<void> {
     'ios:plist',
     'ios:launchScreen',
 
+    // IDE
+    'ide:vscode',
```
Comment (AngeloAvv) [inline lib/processors/processor.dart:81]: I would remove it from the default instructions set since not everyone uses vscode.

Comment (MrTarantula) [reply lib/processors/processor.dart:81]: How do you feel about it being part of the config?:
```yaml
flavorizr:
  app:
    ...
  flavors:
    apple:
      ...
    banana:
      ...
  ide: vscode
```

Comment (AngeloAvv) [reply lib/processors/processor.dart:81]: I was exactly thinking about that. In that case, it should look similar to the dummyasset processor. I would consider it optional and I would use an enum. If that field is missing, then no ide processor is executed.

File: lib/processors/ide/vscode_launch_processor.dart (hunk 1 above)
Comment (MrTarantula) [reply lib/processors/ide/vscode_launch_processor.dart]: Good catch.

File: lib/processors/commons/new_file_string_processor.dart [hunk 3]
```diff
@@ -31,4 +31,10 @@ class NewFileStringProcessor extends AbstractFileStringProcessor {
     String path,
     StringProcessor processor,
   ) : super(path, processor);
+
+  @override
+  void execute() {
+    file.createSync(recursive: true);
+    super.execute();
+  }
```
Comment (AngeloAvv) [inline lib/processors/commons/new_file_string_processor.dart:39]: I think that's not needed. The NewFileStringProcessor doesn't perform an existing check when loaded, so it shouldn't throw any exception.

I should write some dart doc in these basic classes :)

Comment (MrTarantula) [reply lib/processors/commons/new_file_string_processor.dart:39]: It throws an exception when creating a file in a folder that does not exist, in this case `.vscode`.

```
Executing task ide:vscode
Unhandled exception:
FileSystemException: Cannot open file, path = '.vscode/launch.json' (OS Error: The system cannot find the path specified.
, errno = 3)
#0      _File.throwIfError (dart:io/file_impl.dart:645:7)
#1      _File.openSync (dart:io/file_impl.dart:489:5)
#2      _File.writeAsBytesSync (dart:io/file_impl.dart:614:31)
#3      _File.writeAsStringSync (dart:io/file_impl.dart:638:5)
#4      AbstractFileStringProcessor.execute (package:flutter_flavorizr/processors/commons/abstract_file_string_processor.dart:40:15)
#5      Processor.execute (package:flutter_flavorizr/processors/processor.dart:100:23)
<asynchronous suspension>
#6      execute (package:flutter_flavorizr/flutter_flavorizr.dart:43:13)
#7      main (file:///C:/Users/ryan.tauriainen/flutter_flavorizr/bin/flutter_flavorizr.dart:4:3)
#8      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:301:19)
#9      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
pub finished with exit code 255
```

Comment (MrTarantula) [reply lib/processors/commons/new_file_string_processor.dart:39]: If I create the folder before running, it processes it.

Comment (AngeloAvv) [reply lib/processors/commons/new_file_string_processor.dart:39]: I see, so I think we need to create the folder first, and then run the processors.

The ide:vscode command should be a QueueProcessor with a NewFolderProcessor (it doesn't exist, you will have to create it in the commons processor) and a NewFileProcessor with a nested VSCodeLaunchProcessor.

**Comment by MrTarantula** [comment]
Am I on the right track?

**Comment by AngeloAvv** [comment]
Yes, good job @MrTarantula , I'm sorry if I've been to strict over the previous commits, but in the end you did it!

As soon as possible I want to create new sections in the README, one of them about contribution, others people work must be recognized.

**Comment by MrTarantula** [comment]
I understand about being particular about code in your own libraries. I feel the same.

**Pull request merged**